### DISCUSSION TITLE
Building a new module or plug-in

### DISCUSSION THREAD

**Discussion by infotecnica** [joomengine/Joomla-Component-Builder#684] [General]
Hello everyone

Is it possible to build plugin or module and compile it without associating it with a component?

**Comment by Llewellynvdm** [comment]
No not really, but I have seen developers use the demo component to piggyback their modules and plugins on it.

But due to the nature of integration available between components and modules and plugins in JCB. We have chosen to keep that link strong. But with the piggyback option you could build modules and plugins that are independent of the components.

You will just have to make sure that your fields are added as linked to the plugin:
![image](https://user-images.githubusercontent.com/5607939/110129397-95831f00-7dd0-11eb-8e36-a16b21711cb3.png)

You also should change the component owner and stuff to reflect your details in the header licensing part of the files and this is taken from the component.
![image](https://user-images.githubusercontent.com/5607939/110128981-0f66d880-7dd0-11eb-822e-07124d84ecc4.png)

This means yes you can, but only with a piggyback approach :+1:

**Reply by infotecnica** [reply]
Ok Thanks for the full explanation
